src/status_im/contexts/wallet/wallet_connect/responding_events.cljs

```diff
@@ -122,17 +122,25 @@
                             (rf/dispatch [:dismiss-modal screen])
                             (rf/dispatch [:wallet-connect/reset-current-request]))}]]})))
 
+(rf/reg-event-fx
+ :wallet-connect/reject-session-proposal-pure
+ (fn [_ [{:keys [web3-wallet proposal]}]]
+   {:fx [[:effects.wallet-connect/reject-session-proposal
+          {:web3-wallet web3-wallet
+           :proposal    proposal
+           :on-success  #(log/info "Wallet Connect session proposal rejected")
+           :on-error    #(log/error "Wallet Connect unable to reject session proposal")}]
+         [:dispatch [:wallet-connect/reset-current-session-proposal]]]}))
+
 (rf/reg-event-fx
  :wallet-connect/reject-session-proposal
  (fn [{:keys [db]} _]
    (let [web3-wallet      (get db :wallet-connect/web3-wallet)
          current-proposal (get-in db [:wallet-connect/current-proposal :request])]
-     {:fx [[:effects.wallet-connect/reject-session-proposal
-            {:web3-wallet web3-wallet
-             :proposal    current-proposal
-             :on-success  #(log/info "Wallet Connect session proposal rejected")
-             :on-error    #(log/error "Wallet Connect unable to reject session proposal")}]
-           [:dispatch [:wallet-connect/reset-current-session-proposal]]]})))
+     {:fx [[:dispatch
+            [:wallet-connect/reject-session-proposal-pure
+             {:web3-wallet web3-wallet
+              :proposal    current-proposal}]]]})))
```

Comment:
1. The naming is confusing since the `:wallet-connect/reject-session-proposal-pure` event is not a pure function or something like that.
2. I simply don't get the reason behind chaining events — it also affects the performance. If you want to pass proposal while being able to use it for current proposal, why not doing something like that?

```
(let [current-proposal (or proposal (get-in db [:wallet-connect/current-proposal :request]))]
```

Reply:
agreed, we can keep reject-session-proposal and make passing the proposal optional, so if it's not there we use the one from db.

Reply:
I made this version to handle the case where the proposal might not be in DB. 

In the code below, if `(and (not-empty session-networks) required-networks-supported?)` yields `false`, the proposal will not be in the db. It will be nowhere, but we still need to reject it.

```clj
(rf/reg-event-fx
 :wallet-connect/on-session-proposal
 (fn [{:keys [db]} [proposal]]
   (log/info "Received Wallet Connect session proposal: " {:id (:id proposal)})
   (let [accounts                     (get-in db [:wallet :accounts])
         current-viewing-address      (get-in db [:wallet :current-viewing-account-address])
         available-accounts           (filter #(and (:operable? %)
                                                    (not (:watch-only? %)))
                                              (vals accounts))
         networks                     (wallet-connect-core/get-networks-by-mode db)
         session-networks             (wallet-connect-core/proposal-networks-intersection proposal
                                                                                          networks)
         required-networks-supported? (wallet-connect-core/required-networks-supported? proposal
                                                                                        networks)]
     (if (and (not-empty session-networks) required-networks-supported?)
       {:db (update db
                    :wallet-connect/current-proposal assoc
                    :request                         proposal
                    :session-networks                session-networks
                    :address                         (or current-viewing-address
                                                         (-> available-accounts
                                                             first
                                                             :address)))
        :fx [[:dispatch
              [:open-modal :screen/wallet.wallet-connect-session-proposal]]]}
       {:fx [[:dispatch
              [:wallet-connect/session-networks-unsupported proposal]]]}))))

```

Reply:
we can set the proposal in the db in `on-session-proposal`, so reject-session-proposal has access to it regardless of whether it passes the check or not

Reply:
We'll go into a logically odd state where the session is not valid (not supported because of network mismatch), but it's still in the state. 

Having two versions: pure and one-that-reads from state is more clear and straight forward.

Reply:
As long as the event (current-proposal) is cleared up at the end, I see no logical issues with keeping the event in the db even if it's invalid. Do we have any code/functionality that assumes that the presence of the event in the db makes the request valid?

Reply:
> Do we have any code/functionality that assumes that the presence of the event in the db makes the request valid?
Afaik, no we don't have this assumption. I'll combine them back to one event then